apps/ledger-live-desktop/tests/page/swap.page.ts

```diff
@@ -208,4 +208,44 @@ export class SwapPage extends AppPage {
       }
     });
   }
+
+  @step("Select currency to swap from")
+  async selectAssetFrom(electronApp: ElectronApplication, accountToSwapFrom: Account) {
+    const [, webview] = electronApp.windows();
+    await webview.getByTestId(this.fromAccountCoinSelector).click();
+    await this.chooseAssetDrawer.chooseFromAsset(accountToSwapFrom.currency.name);
+  }
+
+  @step("Fill in amount: $1")
+  async fillInOriginCurrencyAmount(electronApp: ElectronApplication, amount: string) {
+    const [, webview] = electronApp.windows();
+    await webview.getByTestId(this.fromAccountAmoutInput).fill(amount);
+  }
+
+  @step("Select currency to swap to: $1")
+  async selectAssetTo(electronApp: ElectronApplication, currency: string) {
+    const [, webview] = electronApp.windows();
+    await webview.getByTestId(this.toAccountCoinSelector).click();
+    await this.chooseAssetDrawer.chooseFromAsset(currency);
+  }
+
+  @step("Verify minimum swap amount error message is displayed")
```

Comment:
[ASK] I am not sure to understand if we test all the scenarii in the tickets (Not enough balance,  Minimum XXXX {Coin} needed for quotes...)

Reply:
basically yes, if `account.accountType` has a value, we fall into [that condition](https://github.com/LedgerHQ/ledger-live/pull/8171/files#diff-cde697e867e613d386476b5883f2a735510bfef5a2c3bcfac77936ac40b06120R246) (on which parent is used to pay the fees), if not we fall into [this condition](https://github.com/LedgerHQ/ledger-live/pull/8171/files#diff-cde697e867e613d386476b5883f2a735510bfef5a2c3bcfac77936ac40b06120R239)

`Minimum XXXX {Coin} needed for quotes...) `assertion is done [here](https://github.com/LedgerHQ/ledger-live/pull/8171/files#diff-cde697e867e613d386476b5883f2a735510bfef5a2c3bcfac77936ac40b06120R240)

atm a few test cases remain to be done, they aren't checked :

- B2CQA-2363: [SWAP] - Error message when network fees are above account balance
- B2CQA-2160: [SWAP] Minimum amount